File: java/Algorithm/src/Custom001.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Custom001 {
    public static void main(String[] args) throws IOException {
        InputStreamReader ir = new InputStreamReader(System.in);
        BufferedReader br = new BufferedReader(ir);

        int t = Integer.parseInt(br.readLine());

        for (int testNum = 1; testNum <= t; testNum++) {
            String[] inputData = br.readLine().split(" ");
            int n = Integer.parseInt(inputData[0]), k = Integer.parseInt(inputData[1]);
            int[] interests = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).sorted().toArray();

            System.out.printf("#%d: %d\n", testNum, solution(n, k, interests));
        }

        br.close();
        ir.close();
    }

    private static int solution(int n, int k, int[] interests) {
        int maxGroup = 0, startIdx = 0, endIdx = 0;

        while (endIdx < n) {
            while (endIdx < n && interests[endIdx] - interests[startIdx] <= k)
                endIdx++;

            maxGroup = Math.max(maxGroup, endIdx - startIdx);

            while (endIdx < n && interests[endIdx] - interests[startIdx] > k)
                startIdx++;
        }

        return maxGroup;
    }
}

File: kotlin/Algorithm/src/main/kotlin/Custom001.kt
import kotlin.math.max

fun main() {
    for (testNum in 1..readln().toInt()) {
        val inputData = readln().split(' ')
        val n = inputData[0].toInt()
        val k = inputData[1].toInt()

        val interests: Array<Int> = readln().split(' ').map { s -> s.toInt() }.toTypedArray()

        println("#${testNum}: ${solution(n, k, interests)}")
    }
}

fun solution(n: Int, k: Int, interests: Array<Int>): Int {
    var maxGroup = 0
    var startIdx = 0
    var endIdx = 0

    interests.sort()

    while (endIdx < n) {
        while (endIdx < n && interests[endIdx] - interests[startIdx] <= k)
            endIdx++

        maxGroup = max(maxGroup, endIdx - startIdx)

        while (endIdx < n && interests[endIdx] - interests[startIdx] > k)
            startIdx++
    }

    return maxGroup
}